internal/pkg/cli/init.go

```diff
@@ -221,6 +227,9 @@ func (o *initOpts) deployEnv() error {
 		// User chose not to deploy the application, exit.
 		return nil
 	}
+	if err := o.initEnv.Ask(); err != nil {
+		return err
+	}
 	return o.initEnv.Execute()
```

Comment:
While trying to understand how come this worked, I realized that we have a bug in `"env init"` :|

Right now the value from `askEnvProfile` in `env_init.go` gets ignored inside `Execute`. So even if you choose a different profile than "default", we pick up only "default".
https://github.com/aws/amazon-ecs-cli-v2/blob/b223142e566e3bd123b541ff8cc500d4b3a113fb/internal/pkg/cli/env_init.go#L120

We should change the code so that we initialize `envDeployer` and `envIdentity` only after the `EnvProfile` value is set.

Here is a sample way of how to do it in `env_delete.go`:
https://github.com/aws/amazon-ecs-cli-v2/blob/b223142e566e3bd123b541ff8cc500d4b3a113fb/internal/pkg/cli/env_delete.go#L60